src/asynk/async_queue.rs

```diff
@@ -381,6 +382,30 @@ mod async_queue_tests {
         assert_eq!(Some("AsyncTask"), type_task);
         transaction.rollback().await.unwrap();
     }
+    #[tokio::test]
+    async fn failed_task_query_test() {
+        let pool = pool().await;
+        let mut connection = pool.get().await.unwrap();
+        let mut transaction = connection.transaction().await.unwrap();
+
+        let task =
+            AsyncQueue::<NoTls>::insert_task_query(&mut transaction, &AsyncTask { number: 1 })
+                .await
+                .unwrap();
+        let metadata = task.metadata.as_object().unwrap();
+        let number = metadata["number"].as_u64();
+        let type_task = metadata["type"].as_str();
+        let id = task.id;
+        assert_eq!(Some(1), number);
+        assert_eq!(Some("AsyncTask"), type_task);
+        let failed_task =
+            AsyncQueue::<NoTls>::fail_task_query(&mut transaction, task, "Some error")
+                .await
+                .unwrap();
+        assert_eq!(id, failed_task.id);
+        assert_eq!(Some("Some error"), failed_task.error_message.as_deref());
```

Comment:
please also check the state

Reply:
okay :D

Reply:
state check commit resolves this conversation